src/ScheduleFunctions.cpp

```diff
@@ -534,6 +540,26 @@ Stmt build_produce(Function f) {
             }
         }
 
+        Stmt annotate;
+        if (target.has_feature(Target::MSAN)) {
+            // Mark the buffers as initialized before calling out.
```

Comment:
I think the vector should at least contain Exprs that are the buffer_t objects needing annotation, so we don't need to plumb the name around and reconstruct the Variables here.